### PR TITLE
CPR-362 call matcher self match score

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/SearchService.kt [hunk 1]
```diff
@@ -38,6 +38,8 @@ class SearchService(
     return matches.firstOrNull()?.candidateRecord
   }
 
+  fun retrieveRecordSelfMatchScore(person: Person): Double = matchService.getSelfMatchScore(person)
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/SearchService.kt:41]: It is here as well as we will need to apply a threshold filter to it in the future

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/MatchScoreClient.kt [hunk 2]
```diff
@@ -38,4 +41,24 @@ class MatchRecord(
...
   val pnc: String? = "",
-)
+) {
+  companion object {
+    fun from(newRecord: Person): MatchRecord {
+      return MatchRecord(
+        firstName = newRecord.firstName,
+        lastname = newRecord.lastName,
+        dateOfBirth = newRecord.dateOfBirth?.toString(),
+        pnc = newRecord.references.firstOrNull { it.identifierType == IdentifierType.PNC }?.identifierValue,
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/MatchScoreClient.kt:51]: person.getPnc() would be more convenient - maybe an extension function would help

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/config/MessagingMultiNodeTestBase.kt [hunk 3]
```diff
@@ -43,6 +43,8 @@ import java.util.UUID
 @ExtendWith(MultiApplicationContextExtension::class)
 abstract class MessagingMultiNodeTestBase : IntegrationTestBase() {
 
+  internal var shouldStubSelfMatchScore: Boolean = true
```
[the remark this answers is not in the document]
Comment (markreesmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/config/MessagingMultiNodeTestBase.kt:46]: What is it used for? It always seems to be true in this code sample

**Review by markreesmoj — APPROVED** [review]
Looks OK, couple of suggestions

**Pull request merged**